src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/AlwaysEncryptedEnclaveProviderUtils.cs

```diff
@@ -0,0 +1,152 @@
+            offset += sizeof(int);
+
+            PublicKey = payload.Skip(offset).Take(publicKeySize).ToArray();
+            offset += publicKeySize;
+
+            PublicKeySignature = payload.Skip(offset).Take(publicKeySignatureSize).ToArray();
+            offset += publicKeySignatureSize;
+        }
+    }
+
+    internal enum EnclaveType
+    {
+        None = 0,
+
+        Vbs = 1,
+
+        Sgx = 2
+    }
+
+    // Contains methods to convert cryptography keys between different formats.
+    internal sealed class KeyConverter
+    {
+        // The RSA public key blob is structured as follows:
+        //     BCRYPT_RSAKEY_BLOB   header
+        //     byte[ExponentSize]   publicExponent      
+        //     byte[ModulusSize]    modulus             
+        private readonly struct RSAPublicKeyBlob
+        {
+            // Size of an RSA public key blob
+            internal static readonly int Size = 539;
+            // Size of the BCRYPT_RSAKEY_BLOB header
+            internal static readonly int HeaderSize = 27;
+            // Size of the exponent (final 3 bytes of the header)
+            internal static readonly int ExponentSize = 3;
+            // Size of the modulus (remaining bytes after the header)
+            internal static readonly int ModulusSize = Size - HeaderSize;
+            internal static readonly int ExponentOffset = HeaderSize - ExponentSize;
+            internal static readonly int ModulusOffset = HeaderSize;
+        }
+
+        // Extracts the public key's modulus and exponent from an RSA public key blob
+        // and returns an RSAParameters object
+        internal static RSAParameters RSAPublicKeyBlobToParams(byte[] keyBlob)
+        {
+            Debug.Assert(keyBlob.Length == RSAPublicKeyBlob.Size,
+                $"RSA public key blob was not the expected length. Actual: {keyBlob.Length}. Expected: {RSAPublicKeyBlob.Size}");
+            return new RSAParameters()
+            {
+                Exponent = keyBlob.Skip(RSAPublicKeyBlob.ExponentOffset).Take(RSAPublicKeyBlob.ExponentSize).ToArray(),
+                Modulus = keyBlob.Skip(RSAPublicKeyBlob.ModulusOffset).Take(RSAPublicKeyBlob.ModulusSize).ToArray()
+            };
+        }
+
+        // The ECC public key blob is structured as follows:
+        //     BCRYPT_ECCKEY_BLOB   header
+        //     byte[KeySize]        X     
+        //     byte[KeySize]        Y         
+        private readonly struct ECCPublicKeyBlob
+        {
+            // Size of an ECC public key blob
+            internal static readonly int Size = 104;
+            // Size of the BCRYPT_ECCKEY_BLOB header
+            internal static readonly int HeaderSize = 8;
+            // Size of each coordinate
+            internal static readonly int KeySize = (Size - HeaderSize) / 2;
+        }
+
+        // Magic numbers identifying blob types
+        private readonly struct KeyBlobMagicNumber
```

Comment:
Can you add some reference in the comment about the origin for this MagicNumber?

Reply:
Added link to docs.